packages/consoledot-api/src/queries/UseUser.ts

```diff
@@ -0,0 +1,68 @@
+import { useQuery } from "@tanstack/react-query";
+import { masQueries } from "../queryKeys";
+import { useChrome } from "@redhat-cloud-services/frontend-components/useChrome";
+import type { KafkaInstanceEnhanced } from "consoledot-api/src/transformers/kafkaRequestToKafkaInstanceEnhanched";
+import { ReadyStatuses, SuspendedStatuses } from "ui-models/src/models/kafka";
+import type { Status } from "ui-models/src/models/kafka";
+
+type User = {
+  username: string;
+  isOrgAdmin: boolean;
+};
+
+export const useUser = () => {
+  const { auth } = useChrome();
+
+  const { data } = useQuery<User>({
+    queryKey: masQueries.user(),
+    queryFn: async () => {
+      const user = await auth.getUser();
+      const username = user?.identity?.user?.username ?? "";
+      const isOrgAdmin = user?.identity?.user?.is_org_admin ?? false;
+
+      if (!username || !isOrgAdmin) {
+        throw new Error("Username or isOrgAdmin is undefined");
+      }
+
+      return {
+        username,
+        isOrgAdmin,
+      };
+    },
+  });
+
+  const defaultUser = {
+    username: "",
+    isOrgAdmin: false,
+    canOpenConnection: () => false,
+    isUserOwnerOrAdmin: () => false,
+    canChangeOwner: () => false,
+  };
+
+  if (!data) {
+    console.warn("Data is undefined");
+    return defaultUser;
+  }
+
+  const canOpenConnection = (instance: KafkaInstanceEnhanced): boolean => {
+    return instance ? ReadyStatuses.includes(instance.status) : false;
+  };
+
+  const isUserOwnerOrAdmin = (owner: string) => {
+    return owner === data.username || data.isOrgAdmin;
+  };
+
+  const canChangeOwner = (owner: string, status: Status) => {
+    return (
+      owner === data.username ||
+      (data.isOrgAdmin && !SuspendedStatuses.includes(status))
+    );
+  };
+
+  return {
+    ...data,
+    canOpenConnection,
+    isUserOwnerOrAdmin,
+    canChangeOwner,
+  };
```

Comment:
Same comment I have made elsewhere. We need to handle the loading off the data, we can't just default all. Add the usare is a critical bit of data, this should be a gate. And all the object construction should happen inside the useQuery, not outside of it.